medicines/doc-index-updater/src/create_manager/mod.rs

```diff
@@ -1,159 +1,139 @@
 
     loop {
-        match try_process_from_queue(&mut create_client).await {
-            Ok(status) => match status {
-                FileProcessStatus::Success(job_id) => {
-                    let _ = state_manager.set_status(job_id, JobStatus::Done).await?;
-                }
-                FileProcessStatus::NothingToProcess => {}
-            },
-            Err(e) => {
-                tracing::error!("{:?}", e);
-            }
+        match try_process_from_queue(&mut create_client, &state_manager).await {
+            Ok(()) => {}
+            Err(e) => tracing::error!("{:?}", e),
         }
         delay_for(time_to_wait).await;
     }
 }
 
-async fn create_file_in_blob(file: Vec<u8>) -> String {
-    format!("create file in blob: {:?}", file)
-}
-
-async fn add_to_search_index(blob: String) {
-    tracing::debug!("update the index for {}", blob);
-}
-
 async fn try_process_from_queue(
-    create_client: &mut DocIndexUpdaterQueue,
-) -> Result<FileProcessStatus, anyhow::Error> {
+    service_bus_client: &mut DocIndexUpdaterQueue,
+    state_manager: &StateManager,
+) -> Result<(), anyhow::Error> {
     tracing::info!("Checking for create messages");
     let retrieved_result: Result<RetrievedMessage<CreateMessage>, RetrieveFromQueueError> =
-        create_client.receive().await;
+        service_bus_client.receive().await;
 
     if let Ok(retrieval) = retrieved_result {
-        let message = retrieval.message.clone();
-        tracing::info!("{:?} message receive!", message);
-        let file = sftp_client::retrieve(
-            message.document.file_source.clone(),
-            message.document.file_path.clone(),
-        )
-        .await?;
-        let _metadata = derive_metadata_from_message(&message.document);
-        let blob = create_file_in_blob(file).await;
-        add_to_search_index(blob).await;
-        retrieval.remove().await?;
-
-        Ok(FileProcessStatus::Success(message.job_id))
-    } else {
-        Ok(FileProcessStatus::NothingToProcess)
+        let processing_result = process_message(retrieval.message.clone()).await;
+        match processing_result {
+            Ok(FileProcessStatus::Success(job_id)) => {
+                let _ = state_manager.set_status(job_id, JobStatus::Done).await?;
+            }
+            Ok(FileProcessStatus::NothingToProcess) => {}
+            Err(e) => {
+                if e.to_string()
+                    == "Couldn't retrieve file: [-31] Failed opening remote file".to_string()
+                {
+                    tracing::info!("Updating state to errored and removing message");
+                    let _ = state_manager.set_status(
+                        retrieval.message.job_id,
+                        JobStatus::Error {
+                            message: "Couldn't find file".to_string(),
+                            code: "404".to_string(),
+                        },
+                    );
```

Comment:
@TimboTambo I've been trying to follow this example, but I'm not sure it works.

From what I understand here, `_` will be a future thingy, and once you `await` it, you get an error with the message "Status unknown: Error(: Couldn't find file)".

Let's chat about it tomorrow.